src/compiler/scanner/lines_1000_2000.rs

```diff
@@ -23,44 +30,188 @@ impl Scanner {
+            }
+        } else {
+            result = self.text_substring(start, end);
+        }
 
-        if false {
-            unimplemented!()
+        if decimal_fragment.is_some() || self.token_flags().intersects(TokenFlags::Scientific) {
+            self.check_for_identifier_start_after_numeric_literal(
+                on_error,
+                start,
+                Some(
+                    decimal_fragment.is_none()
+                        && self.token_flags().intersects(TokenFlags::Scientific),
+                ),
+            );
+            ScanNumberReturn {
+                type_: SyntaxKind::NumericLiteral,
+                value: result.parse::<f64>().unwrap().to_string(),
+            }
         } else {
             self.set_token_value(result);
             let type_ = self.check_big_int_suffix();
+            self.check_for_identifier_start_after_numeric_literal(on_error, start, None);
             ScanNumberReturn {
                 type_,
                 value: self.token_value(),
             }
         }
     }
 
+    pub(super) fn check_for_identifier_start_after_numeric_literal(
+        &self,
+        on_error: Option<ErrorCallback>,
+        numeric_start: usize,
+        is_scientific: Option<bool>,
+    ) {
+        let is_scientific = is_scientific.unwrap_or(false);
+        if !is_identifier_start(
+            code_point_at(self.text(), self.pos()),
+            Some(self.language_version),
+        ) {
+            return;
+        }
+
+        let identifier_start = self.pos();
+        let length = self.scan_identifier_parts(on_error).len();
+
+        if length == 1 && self.text_char_at_index(identifier_start) == CharacterCodes::n {
+            if is_scientific {
+                self.error(
+                    on_error,
+                    &Diagnostics::A_bigint_literal_cannot_use_exponential_notation,
+                    Some(numeric_start),
+                    Some(identifier_start - numeric_start + 1),
+                );
+            } else {
+                self.error(
+                    on_error,
+                    &Diagnostics::A_bigint_literal_must_be_an_integer,
+                    Some(numeric_start),
+                    Some(identifier_start - numeric_start + 1),
+                );
+            }
+        } else {
+            self.error(
+                on_error,
+                &Diagnostics::An_identifier_or_keyword_cannot_immediately_follow_a_numeric_literal,
+                Some(identifier_start),
+                Some(length),
+            );
+            self.set_pos(identifier_start);
+        }
+    }
+
+    pub(super) fn scan_octal_digits(&self) -> u32 {
+        let start = self.pos();
+        while matches!(self.maybe_text_char_at_index(self.pos()), Some(ch) if is_octal_digit(ch)) {
+            self.increment_pos();
+        }
+        u32::from_str_radix(&self.text_substring(start, self.pos()), 8).unwrap()
```

Comment:
Here again the Typescript version uses `+` coercion, using `u32::from_str_radix()` should work for converting an octal/binary string into the corresponding number value